pom.xml

```diff
@@ -100,6 +101,34 @@
             </properties>
         </profile>
         <profile>
+            <!-- maven-checkstyle-plugin required Java 7 or later -->
+            <id>java-7-or-later-profile</id>
+            <activation>
+                <jdk>[1.7,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <version>${maven-checkstyle-plugin.version}</version>
+                    <configuration>
+                    <failsOnError>true</failsOnError>
+                        <configLocation>google_checks.xml</configLocation>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <phase>verify</phase>
+                            <goals>
+                                <goal>checkstyle</goal>
```

Comment:
shouldnt we use `check` goal here instead? As taken from documentation:
* *checkstyle:checkstyle* is a reporting goal that performs Checkstyle analysis and generates a report on violations.
* *checkstyle:check* is a goal that performs Checkstyle analysis and outputs violations or a count of violations to the console, potentially failing the build. It can also be configured to re-use an earlier analysis.